extension/src/popup/components/accountHistory/HistoryItem/index.tsx

```diff
@@ -112,169 +109,229 @@ export const HistoryItem = ({
     setIsDetailViewShowing,
   };
 
-  if (isSwap) {
-    PaymentComponent = (
-      <>
-        {new BigNumber(amount).toFixed(2, 1)} {destAssetCode}
-      </>
-    );
-    rowText = t(`{{srcAssetCode}} for {{destAssetCode}}`, {
-      srcAssetCode,
-      destAssetCode,
-    });
-    dateText = t(`Swap \u2022 {{date}}`, { date });
-    transactionDetailProps = {
-      ...transactionDetailProps,
-      headerTitle: t(`Swapped {{srcAssetCode}} for {{destAssetCode}}`, {
-        srcAssetCode,
-        destAssetCode,
-      }),
-      operationText: `+${new BigNumber(amount)} ${destAssetCode}`,
-    };
-  } else if (isPayment) {
-    // default to Sent if a payment to self
-    isRecipient = to === publicKey && from !== publicKey;
-    paymentDifference = isRecipient ? "+" : "-";
-    PaymentComponent = (
-      <>
-        {paymentDifference}
-        {formatAmount(new BigNumber(amount).toFixed(2, 1))} {destAssetCode}
-      </>
-    );
-    IconComponent = isRecipient ? (
-      <Icon.ArrowDown className="HistoryItem__icon--received" />
-    ) : (
-      <Icon.ArrowUp className="HistoryItem__icon--sent" />
-    );
-    rowText = destAssetCode;
-    dateText = `${isRecipient ? t("Received") : t("Sent")} \u2022 ${date}`;
-    transactionDetailProps = {
-      ...transactionDetailProps,
-      isRecipient,
-      headerTitle: `${
-        isRecipient ? t("Received") : t("Sent")
-      } ${destAssetCode}`,
-      operationText: `${paymentDifference}${new BigNumber(
-        amount,
-      )} ${destAssetCode}`,
-    };
-  } else if (isCreateExternalAccount) {
-    PaymentComponent = <>-{new BigNumber(startingBalance).toFixed(2, 1)} XLM</>;
-    IconComponent = <Icon.ArrowUp className="HistoryItem__icon--sent" />;
-    rowText = "XLM";
-    dateText = `${t("Sent")} \u2022 ${date}`;
-    transactionDetailProps = {
-      ...transactionDetailProps,
-      headerTitle: t("Create Account"),
-      isPayment: true,
-      operation: {
-        ...operation,
-        asset_type: "native",
-        to: account,
-      },
-      operationText: `-${new BigNumber(startingBalance)} XLM`,
-    };
-  } else if (isSorobanTx) {
-    const attrs = getAttrsFromSorobanHorizonOp(operation, networkDetails);
-    const token = tokenBalances.find(
-      (balance) => attrs && balance.contractId === attrs.contractId,
-    );
+  const [txDetails, setTxDetails] = React.useState(transactionDetailPropsBase);
```

Comment:
Super NIT: but in other files in the repo, we usually just `import { useState } from "react"` at the top of the file so we just call `useState` directly instead of calling `React.useState`. So minor, but would be good to keep the pattern consistent

Reply:
Same deal with `useEffect`

Reply:
ah whoops, yup will update.

Reply:
done in 1ba0450fb6a466570f2ab0318b838f1072f0ae47